zboxcore/allocationchange/moveobject.go

```diff
@@ -50,9 +54,14 @@ func (ch *MoveFileChange) ProcessChange(rootRef *fileref.Ref) error {
 	}
 
 	if dirRef.GetPath() != ch.DestPath || dirRef.GetType() != fileref.DIRECTORY {
-		return errors.New("file_not_found", "Object to move not found in blobber")
+		err = errors.New("file_not_found", "Object to move not found in blobber")
+		return
 	}
 
+	commitParam.WmFileID = dirRef.FileID
+	commitParam.LatestFileID = latestFileID
```

Comment:
No we should not.
FileID is the only immutable field of a file. Other all fields will change upon file operations.
FileID is crucial for repairing. While moving file we don't add any files so there is fixed inode(or latest inode).

But your concern regarding source path and destination path is valid. For repair process we need to know both source and destination fileID.
Paths can be determined from file ref belonging to fileID itself.

After Tom Austin will validate the addition of sourceID field in writemarker, it will applied.